### DISCUSSION TITLE
how to configure nextjs to see tsconfig.build.json instead of jsconfig.json when build?

### DISCUSSION THREAD

**Discussion by walt1992** [vercel/next.js#17196] [Help] [answered]
I have two config files for TypeScript. One is general `tsconfig.json` for the IDE, and another one is `tsconfig.build.json` to be used in real build. However, I don't know how to configure nextjs to use tsconfig.build.json file when it run on build. 

Is there way to set for it?

**Comment by antonio-development** [comment]
It is actual! We need separate configs  tsconfig.json for the IDE, and another one is tsconfig.build.json to be used in real build.

By the way, the only one way solve this problem temporarily is to disable type checking for production.
By adding this to the production config in next.config.js.
```js
    typescript: {
      // !! WARN !!
      // Dangerously allow production builds to successfully complete even if
      // your project has type errors.
      // !! WARN !!
      ignoreBuildErrors: true,
    },
```
And optionally disable linting in package.json
```json
"build": "next build --no-lint",
```

**Comment by sergiop** [comment]
You can simply rename the `tsconfig.build.json` in `tsconfig.json` before running the build script.

If you run the build script in the CI, you can run something like:
```yaml
script:
  - yarn install
  - mv tsconfig.build.json tsconfig.json
  - yarn build
```

[EDIT]: I know, it's not fancy, but it works.

**Comment by icyJoseph** [comment, marked as answer]
There's a typescript option the `next.config.js` file:

```js
  typescript: {
    tsconfigPath: './tsconfig.wonky.json',
  },
```

For example the above will look for `'./tsconfig.wonky.json'` relative to the `next.config.js` file.

- https://nextjs.org/docs/app/api-reference/config/typescript#custom-tsconfig-path

**Reply by Staslios** [reply]
Thanks man, you saved my time! Is this option documented somewhere ? The only thing I can find is [this link](https://nextjs.org/docs/app/api-reference/next-config-js/typescript).

**Reply by romgrk** [reply]
Why is it not documented.

**Reply by rvanlaak** [reply]
Landed here through search; it now seems documented: https://nextjs.org/docs/app/api-reference/config/typescript#custom-tsconfig-path

**Comment by ellis** [comment]
The above solution works for me.  Using next v14.1.0.  Thanks, @icyJoseph!
I'd say this question is answered.

**Comment by antspy** [comment]
Is there any official documentation on this `tsconfigPath` solution? 

It seems to have no effect at the moment, even setting that path to a non-existent file does not seem to raise any issues during compilation

**Reply by icyJoseph** [reply]
Are you nesting it within the typescript field? I see it is still around in the codebase. A lot of the next.config options are best seen in the source code, for better or worse

**Reply by antspy** [reply]
Thank you for your comment! 

If you could provide a link to the right place to look at in the repo, that would be great - so I can try to explore it a bit more and understand it better :)
